js/slidedeck.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * A slide deck object
+ */
+class SlideDeck {
+  /**
+   * Constructor for the SlideDeck object.
+   * @param {NodeList} slides A list of HTML elements containing the slide text.
+   * @param {L.Map} map The Leaflet map where data will be shown.
+   * @param {Object} layerOptions An object containing styling choices for the layers.
+   */
+  constructor(slides, map, layerOptions) {
+    this.slides = slides;
+    this.map = map;
+    this.layerOptions = layerOptions;
+
+    this.dataLayer = L.layerGroup().addTo(map);
+    this.currentSlideIndex = 0;
+  }
+
+  /**
+ * ### updateDataLayer
+ *
+ * Clears previous markers or shapes, and replaces them with the new GeoJSON data.
+ * @param {object} data A GeoJSON FeatureCollection object.
+ * @param {object} [options] Optional parameter that specifies layer styling and other settings.
+ * @return {L.GeoJSONLayer} The new GeoJSON layer added to the data layer group.
+ */
+  updateDataLayer(data, options) {
+    this.dataLayer.clearLayers();
+    const geoJsonLayer = L.geoJSON(data, options || {
+      pointToLayer: (p, latlng) => L.marker(latlng),
+      style: (feature) => {
+        return {
+          fillColor: 'red',
+        };
+      },
+    })
+        .bindTooltip((l) => l.feature.properties.label)
```

Comment:
**suggestion:** You're binding tooltips to your layers in index.js (which is great), but if you get rid of this line you'll also get rid of a bunch of errors in the console.
```suggestion
```